source/Graphite.System/Kernel.cs

```diff
@@ -141,7 +141,8 @@ private Action CreateReportingAction(CounterListenerElement config)
 
         private Action CreateReportingAction(AppPoolElement config, out AppPoolListener listener)
         {
-            var element = new AppPoolListener(config.AppPoolName);
+
+            var element = config.WorkingSet && string.IsNullOrEmpty(config.Counter) ? new AppPoolListener(config.AppPoolName) : new AppPoolListener(config.AppPoolName, config.Category, config.Counter);
```

Comment:
For old configurations with `workingSet="false"` a new counter with empty "Category" and "Counter" would be created here...